frontend/src/pages/AuthCallbackPage.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useEffect, useRef } from 'react';
+import { useNavigate, useSearchParams } from 'react-router-dom';
+import { message } from 'antd';
+import {KEYCLOAK_CLIENT_ID, KEYCLOAK_TOKEN_URL} from "../consts/apiConsts.ts";
+
+const AuthCallbackPage: React.FC = () => {
+    const [searchParams] = useSearchParams();
+    const navigate = useNavigate();
+    const hasHandledCallback = useRef(false);
+
+    useEffect(() => {
+        const handleAuthCallback = async () => {
+            if (hasHandledCallback.current) {
+                return;
+            }
+            hasHandledCallback.current = true;
+
+            const code = searchParams.get('code');
+            if (!code) {
+                message.error('No authorization code in URL');
+                navigate('/login');
+                return;
+            }
+
+            try {
+                console.log('Code:', code);
+                console.log('Redirect URI:', `${window.location.origin}/auth/callback`);
+
+                const response = await fetch(`${KEYCLOAK_TOKEN_URL}`, {
+                    method: 'POST',
+                    headers: {
+                        'Content-Type': 'application/x-www-form-urlencoded',
+                    },
+                    body: new URLSearchParams({
+                        grant_type: 'authorization_code',
+                        code: code,
+                        redirect_uri: `${window.location.origin}/auth/callback`,
+                        client_id: `${KEYCLOAK_CLIENT_ID}`,
+                    }),
+                });
```

Comment:
It would probably be better to use axios.

Reply:
And here, for example, there would be a problem with returning errors from the interceptor, because logging in does not have a token

Reply:
Done